cpp/src/plasma/CMakeLists.txt

```diff
@@ -22,6 +22,25 @@ project(plasma VERSION "${ARROW_BASE_VERSION}")
 
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/../python/cmake_modules")
 
+# Include jemalloc
+
+# TODO(pcm): Include xhochy's patch
+set(JEMALLOC_PREFIX "${CMAKE_CURRENT_BINARY_DIR}/jemalloc_ep-prefix/src/jemalloc_ep/dist/")
+set(jemalloc_URL https://github.com/jemalloc/jemalloc/archive/5.1.0.tar.gz)
+set(JEMALLOC_INCLUDE_DIR "${JEMALLOC_PREFIX}/include")
+set(JEMALLOC_SHARED_LIB "${JEMALLOC_PREFIX}/lib/libjemalloc${CMAKE_SHARED_LIBRARY_SUFFIX}")
+set(JEMALLOC_STATIC_LIB "${JEMALLOC_PREFIX}/lib/libjemalloc_pic${CMAKE_STATIC_LIBRARY_SUFFIX}")
+
+ExternalProject_Add(jemalloc
+    URL ${jemalloc_URL}
+    DOWNLOAD_DIR "${DOWNLOAD_LOCATION}"
+    CONFIGURE_COMMAND ./autogen.sh "--prefix=${JEMALLOC_PREFIX}" "--with-jemalloc-prefix=je_plasma_" "--with-private-namespace=je_plasma_private_" "--disable-tls"
```

Comment:
Please use the same jemalloc version as we do in Arrow C++. Eventhough they should live happily along eachother, we should not add multiple versions of a library to the project.

Reply:
Aside: if we're able to remove our vendored jemalloc by making this upgrade that would be great